cmake/partition_manager.cmake

```diff
@@ -378,6 +380,51 @@ else()
       list(APPEND pm_out_region_file ${${d}_PM_DOMAIN_REGIONS})
       list(APPEND global_hex_depends ${${d}_PM_DOMAIN_DYNAMIC_PARTITION}_subimage)
       list(APPEND domain_hex_files ${${d}_PM_HEX_FILE})
+
+      # Add domain prefix cmake variables for all partitions
+      # Here, we actually overwrite the already imported kconfig values
+      # for our own domain. This is not an issue since all of these variables
+      # are accessed through the 'partition_manager' target, and most likely
+      # through generator expression, as this file is one of the last
+      # cmake files executed in the configure stage.
+      import_kconfig(PM_ ${${d}_PM_DOTCONF_FILES} ${d}_pm_var_names)
+
+      foreach(name ${${d}_pm_var_names})
+        set_property(
+          TARGET partition_manager
+          PROPERTY ${d}_${name}
+          ${${name}}
+          )
+      endforeach()
+
+      if (CONFIG_BT_RPMSG_NRF53
+          AND CONFIG_BOOTLOADER_MCUBOOT
+          AND CONFIG_SOC_NRF5340_CPUAPP
+          AND CONFIG_HCI_RPMSG_BUILD_STRATEGY_FROM_SOURCE)
+          # Create symbols for the offset reqired for moving the signed network
+          # core application to MCUBoots secondary slot. This is needed
+          # because  objcopy does not support arithmetic expressions as argument
+          # (e.g. '0x100+0x200'), and all of the symbols used to generate the
+          # offset is only available as a generator expression when MCUBoots
+          # cmake code exectues.
+
+          get_target_property(
+            net_app_addr
+            partition_manager
+            CPUNET_PM_APP_ADDRESS
+            )
+
+          # There is no padding in front of the network core application.
+          math(EXPR net_app_TO_SECONDARY
+            "${PM_MCUBOOT_SECONDARY_ADDRESS} - ${net_app_addr} + ${PM_MCUBOOT_PAD_SIZE}")
+
+          set_property(
+            TARGET partition_manager
+            PROPERTY net_app_TO_SECONDARY
```

Comment:
This comment is not directly related to this line, but I notice that in this PR, several new properties are being defined, like:
- `net_app_TO_SECONDARY`
- `CPUNET_PM_APP_ADDRESS`
- foreach(name ${${d}_pm_var_names}) ? Who knows what goes in here.

and probably more.
The `partition_manager` target already have several properties, so there are increasing risk that new properties are simply being added where an existing property was actually usable or could be made more generic.

With more properties, then there is an increased risk of code copying with small adjustment for new property.

For a start, I would like to see a description in beginning of this file which list the properties and their purpose, for inspiration, look here:
https://github.com/zephyrproject-rtos/zephyr/blob/5d2c3b06eab6f135eac3924365ccf55372572947/cmake/bintools/bintools_template.cmake#L7-L59

Those are all properties, but with an overview of their meaning / usage.

Reply:
Created a JIRA task for it NCSDK-6233